queue/ring.go

```diff
@@ -49,8 +49,14 @@ type nodes []*node
 // described here: http://www.1024cores.net/home/lock-free-algorithms/queues/bounded-mpmc-queue
 // with some minor additions.
 type RingBuffer struct {
-	nodes                          nodes
-	queue, dequeue, mask, disposed uint64
+	buffer0        [8]uint64
+	queue          uint64
+	buffer1        [8]uint64
+	dequeue        uint64
+	buffer2        [8]uint64
+	mask, disposed uint64
+	buffer3        [8]uint64
+	nodes          nodes
```

Comment:
You can use `_` in struct definitions to be explicit about padding (I've seen it used in some C bindings)